R/apa_print_anova.R

```diff
@@ -415,6 +416,30 @@ apa_print.anova <- function(
         , simplify = TRUE
       )
     )
+  # stats::anova.glm() and car::Anova.glm
+  } else if(any(grepl("Deviance", object_heading))) {
+    x$Term <-rownames(x)
+    y <- canonize(x)
+    y <- remove_residuals_row(y)
+    if(all(colnames(x) != "F values")) y$df.residual <- NULL
+    if(any(colnames(x) == "Cp")) y$df <- NULL
+
+    if(is.null(y$statistic)) {
```

Comment:
Hmm, this I don't understand. Shouldn't `canonize` do this?

Reply:
I would consider these fixes specific to analysis of deviance that do not fit into a generic function such as `canonize()` -- moving this to `canonize()` would imply that we would need a list of test statistics that should be reported with degrees of freedom and a list of test statistics that should not be reported with degrees of freedom.

Reply:
I'm not sure I understand. This section does not deal with degrees of freedom, as far as I can tell. I'm referring here to the section where `statistic` is null and is replaced by `deviance`.

Reply:
I accidentally deleted my last comment.

Maybe the most concise answer is the following: In one very specific case (chi-squared tests for generalized linear models), a column named `Deviance` contains the *difference in deviance* which is distributed as chi-squared, and is used as the test statistic.

In `canonize()`, however, we rename `Deviance` to `deviance`, not `statistic`, because columns of this name typically do not contain the test statistic, but simply the deviance of the model (in addition to the test statistic contained in another column). Because column `Deviance` containing the test statistic is the exception, not the rule, I would opt for sticking with the current behavior of `canonize()` and deal with the exception within the specific method.

Reply:
What a mess. That makes sense, thanks for clarifying!